### DISCUSSION TITLE
Spatie\LaravelData\Support\Validation\ValidationContext::__construct(): Argument #1 ($payload) must be of type ?array, string given, called in /var/www/application/vendor/spatie/laravel-data/src/Resolvers/DataValidationRulesResolver.php on line 204

### DISCUSSION THREAD

**Discussion by dima-stazzle** [spatie/laravel-data#351] [Q&A] [answered]
laravel-data version 3 is used.

How to reproduce.

Main DTO file
```
<?php
declare(strict_types=1);

namespace App\DTO;

use Spatie\LaravelData\Data;

class CreationRequest extends Data
{
    public string $name;

    public Address $address;

    public static function rules(): array
    {
        return [
            'name' => 'required|string',
            'address' => 'required|array:street,postcode,city',
        ];
    }
}
```

Nested Address DTO
```
<?php
declare(strict_types=1);

namespace App\DTO;

use Spatie\LaravelData\Data;

class Address extends Data
{
    public string $street;

    public string $postcode;

    public string $city;
}
```

The action in the controller looks like this
```
    public function store(CreationRequest $requestDTO): Response
    {
        .....
    }
```

If we send a request with data like this
```
        [
            'name' => 'Test name ',
            'address' => 'Some address here'
        ];
```
then there will be an error `Spatie\LaravelData\Support\Validation\ValidationContext::__construct(): Argument #1 ($payload) must be of type ?array, string given, called in /var/www/application/vendor/spatie/laravel-data/src/Resolvers/DataValidationRulesResolver.php on line 204`.

Yes, an address field in the request is wrong, it shouldn't be a string and should be an array, but if a client sent a string to a server, there is definitely should not be an error.

In the laravel-data version 2 a rule "'address' => 'required|array:street,postcode,city'," applied and there would be a validation error like "The address must be an array"

Shortly, it is not possible to work with nested objects, because if there is not an array for this object property in the request, then there will be a crash.

**Comment by rubenvanassche** [comment, marked as answer]
This should be fixed in the next release.

**Reply by dima-stazzle** [reply]
It's fixed. Thank you
